lib/ide/parser/src/jsclient.rs

```diff
@@ -53,10 +50,13 @@ impl Client {
 }
 
 impl IsParser for Client {
-    fn parse(&mut self, _program: String) -> api::Result<api::Ast> {
-        match parse(_program) {
-            Ok(json_ast) => Err(ParsingError(json_ast)),
-            Err(_)       => Err(api::interop_error(Error::ScalaException())),
-        }
-    }
-}
+   fn parse(&mut self, program: String, ids: IDMap) -> api::Result<api::Ast> {
+       let json_ids = serde_json::to_string(&ids).expect(
+         "ID Map serialization should not fail."
```

Comment:
Truly should not happen... but if we already return Resutl, we can include this failure as a case in Error type and pass it to the caller.